src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs

```diff
@@ -137,13 +137,19 @@ private static ISignatureProvider GetSignatureProvider(string timestamper)
             try
             {
                 await SigningUtility.SignAsync(signingOptions, signRequest, token);
+
+                if (File.Exists(signingOptions.OutputFilePath))
+                {
+                    FileUtility.Replace(signingOptions.OutputFilePath, packageOutputPath);
+                }
             }
-            finally
+            catch
```

Comment:
apart from throwing the exception, functionally both of these should be equivalent, right?

Reply:
Yes, I decided to change it to a `catch` because in the `try` we are already doing a `replace`, therefore a `delete` won't ever be necessary... The `delete` in the catch is to clean up the remaining temp file.